src/core/Akka/Dispatch/ChannelSchedulerExtension.cs

```diff
@@ -0,0 +1,389 @@
+﻿using Akka.Actor;
+using Akka.Configuration;
+using Akka.Event;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Akka.Dispatch
+{
+    public sealed class ChannelTaskSchedulerProvider : ExtensionIdProvider<ChannelTaskScheduler>
+    {
+        public override ChannelTaskScheduler CreateExtension(ExtendedActorSystem system)
+        {
+            return new ChannelTaskScheduler(system);
+        }
+    }
+
+    public sealed class ChannelTaskScheduler : IExtension, IDisposable
+    {
+        [ThreadStatic]
+        private static TaskSchedulerPriority _threadPriority = TaskSchedulerPriority.None;
+
+        private readonly Task _controlTask;
```

Comment:
For instance, what is the `_controlTask` doing? How does it relate to the `Timer` and the channels themselves?